lib/pacemaker/pcmk_sched_colocation.c

```diff
@@ -45,48 +48,47 @@ cmp_dependent_priority(gconstpointer a, gconstpointer b)
         return -1;
     }
 
-    CRM_ASSERT(rsc_constraint1->dependent != NULL);
-    CRM_ASSERT(rsc_constraint1->primary != NULL);
+    CRM_ASSERT((dependent1 != NULL) && (rsc_constraint1->primary != NULL));
```

Comment:
Out of scope but we should be checking `dependent2` as well (same for the primary cmp function too). We dereference it.

Reply:
Coverity failed on this because we dereference `rsc_constraint{1,2}` before NULL-checking